src/js/game/Entities/Agent.js

```diff
@@ -0,0 +1,156 @@
+const BaseEntity = require("./BaseEntity.js");
+const CallbackCommand = require("../CommandQueue/CallbackCommand.js");
+
+module.exports = class Player extends BaseEntity {
```

Comment:
There's a lot of duplicated functionality here from Player.js.  Can you have `class Agent extends Player` and only override the methods that behave differently?

Reply:
Yeah, that's the end goal. At the moment, I just copied it over with a different tag, trying to get both entities to behave at the same time. A clean up would absolutely entail extending Player.js and then only overloading necessary functions.

I would expect that to be done before the PR is accepted.

Reply:
Seconded that this should definitely be a subclass